crates/astria-sequencer/src/app.rs

```diff
@@ -931,6 +1059,37 @@ impl App {
     }
 }
 
+// updates the priority of the txs in the mempool based on the current state,
+// and removes any txs that are now invalid.
+async fn update_mempool_after_finalization<S: StateReadExt>(
+    mempool: Arc<Mutex<BasicMempool>>,
+    state: S,
+) -> anyhow::Result<()> {
```

Comment:
I believe that as it stands, the mempool could be completely cleared. All transactions left in mempool after execution will get rechecked by cometbft after `commit`. 

I think that without adding a map or something of failed executions, there's little reason to keep it around, and even with that we are could clear out the queue. We are duplicating DB reads, since `CheckTx` will do a state read on nonce again, and then the insert call will do a replace. Making this loop largely duplicate work.

Reply:
good point, i left this because `recheck` can be disabled, even though it's on by default. if we can enforce that `recheck` is always on then that would be ideal, i can remove the nonce updating for now.

 i'll leave the "removed included txs from mempool" part, and instead of clearing the mempool we can do `change_priority` in `check_tx` if the priority changes (and remove tx if it fails ofc), `change_priority` is more efficient that removing and re-adding, especially if there's potentially a lot of txs.